velox/exec/tests/HashJoinTest.cpp

```diff
@@ -69,6 +69,46 @@ class HashJoinTest : public HiveConnectorTestBase {
         params, [](auto*) {}, referenceQuery, duckDbQueryRunner_);
   }
 
+  // This function sets up a hashJoin between tables with different schema.
+  // For the purpose of simplicity, it is assumed that the key columns
+  // `numJoinKeys` appear at the beginning of the tables.
+  void testJoinTablesDifferentSchemas(
+      const std::vector<TypePtr>& leftColTypes,
+      const std::vector<TypePtr>& rightColTypes,
+      int32_t numJoinKeys,
```

Comment:
This function is called only once and the parameters are fixed. It might be simpler to write the new test without an extra indirection of a helper function.